### PR TITLE
Temp fix for application listing error, if the app owner is deleted 

### PR DESCRIPTION
**Pull request by AnuradhaSK**
### Proposed changes in this pull request

$Subject

### CODE REVIEW

**Review by sadilchamishka — APPROVED** [review]

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java [hunk 1]
```diff
@@ -915,20 +915,52 @@ private static User getUserFromTenant(String username, String userId, int tenant
...
+            }
+            Optional<User> user = getUser(tenantDomain, username);
+            if (!user.isPresent()) {
+                /*
+                This situation occur when the application creator is deleted. At that point,
+                 set the tenant domain of the application as the user's tenant domain.
+                 */
+                return tenantDomain;
+            }
+            return user.get().getTenantDomain();
```
Comment (wenodpathirana) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:938]:
```suggestion
            /*
             Else situation occur when the application creator is deleted. At that point,
             set the tenant domain of the application as the user's tenant domain.
             */
            return getUser(tenantDomain, username).map(User::getTenantDomain).orElse(tenantDomain);
   
```

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java [hunk 2]
```diff
@@ -915,20 +915,52 @@ private static User getUserFromTenant(String username, String userId, int tenant
...
+        as the user's tenant domain.
+         */
+        int tenantID = IdentityTenantUtil.getTenantId(tenantDomain);
+        Tenant tenant = ApplicationManagementServiceComponentHolder.getInstance().getRealmService()
+                .getTenantManager().getTenant(tenantID);
+        String accessedOrganizationId = tenant.getAssociatedOrganizationUUID();
+        if (accessedOrganizationId == null) {
+            return true;
+        }
+        return false;
```
Comment (wenodpathirana) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:963]: return StringUtils.isEmpty(accessedOrganizationId);

**Review by wenodpathirana — APPROVED** [review]

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java (hunk 1 above)
Comment (AnuradhaSK) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java]: fixed in [8db456c](https://github.com/wso2/carbon-identity-framework/pull/4302/commits/8db456c974f555f629f9695bce1d362bdedc277f)

**Comment by AnuradhaSK** [comment]
https://github.com/wso2/carbon-identity-framework/pull/4302#issuecomment-1284053611 failed due to "UserMeApprovalTest" failures. Which is a known intermittent test failure. Hence merging

**Pull request merged**